mlir/lib/Dialect/Vector/Transforms/LowerVectorMultiReduction.cpp

```diff
@@ -461,6 +471,31 @@ struct OneDimMultiReductionToTwoDim
     return success();
   }
 };
+
+struct LowerVectorMultiReductionPass
+    : public vector::impl::LowerVectorMultiReductionBase<
+          LowerVectorMultiReductionPass> {
+  LowerVectorMultiReductionPass(vector::VectorMultiReductionLowering option) {
+    this->loweringStrategy = option;
+  }
+
+  void runOnOperation() override {
+    Operation *op = getOperation();
+    MLIRContext *context = op->getContext();
+
+    RewritePatternSet loweringPatterns(context);
+    populateVectorMultiReductionLoweringPatterns(loweringPatterns,
+                                                 this->loweringStrategy);
```

Comment:
Instead of using a pattern, can we extract the pattern body in a utility function and just do a `walk()` here and call the utility?

Reply:
@joker-eph interesting, FMI is this for code simplicity / fewer dependencies or something else?

Reply:
This is for compile-time aspects: the greedy rewriter is expensive. Using it when you have a single pattern to apply and won't benefit from the complexity coming from the greedy rewriter is just a bit sad.

Reply:
Hi @joker-eph,

I understand this point, and I've learned something new from it. Regarding the specific pattern in question, it's actually not just a single pattern but rather a collection of patterns, which includes:

- InnerOuterDimReductionConversion
- ReduceMultiDimReductionRank
- OneDimMultiReductionToTwoDim
- TwoDimMultiReductionToReduction
- TwoDimMultiReductionToElementWise

As you have mentioned, these patterns might not make the time spent in the greedy rewriter stand out. Therefore, I think we can maintain the current implementation here.